Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndoUtils.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/std/string/string.h>
+#include <AzToolsFramework/Prefab/PrefabDomTypes.h>
+
+namespace AzToolsFramework
+{
+    namespace Prefab
+    {
+        namespace PrefabUndoUtils
+        {
+            void GenerateAddEntityPatch(
+                PrefabDom& patch,
+                PrefabDom& newEntityDom,
```

Comment:
nit: This should be no harm to change to `const PrefaomValue&` if I want to pass in a prefab dom value later.